bindings/java/src/org/sleuthkit/datamodel/HostManager.java

```diff
@@ -0,0 +1,170 @@
+ * Contact: carrier <at> sleuthkit <dot> org
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.sleuthkit.datamodel;
+
+import com.google.common.base.Strings;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.HashSet;
+import java.util.Optional;
+import java.util.Set;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import org.sleuthkit.datamodel.SleuthkitCase.CaseDbConnection;
+import org.sleuthkit.datamodel.SleuthkitCase.CaseDbTransaction;
+
+/**
+ * Responsible for creating/updating/retrieving Hosts.
+ *
+ */
+public class HostManager {
+	
+	private static final Logger LOGGER = Logger.getLogger(HostManager.class.getName());
+
+	private final SleuthkitCase db;
+
+	/**
+	 * Construct a HostManager for the given SleuthkitCase.
+	 *
+	 * @param skCase The SleuthkitCase
+	 *
+	 */
+	HostManager(SleuthkitCase skCase) {
+		this.db = skCase;
+	}
+	
+	
+	/**
+	 * Get or create host with specified name.
+	 * 
+	 * @param name	Host name.
+	 * @param transaction Database transaction to use.
+	 * 
+	 * @return
+	 * @throws TskCoreException 
+	 */
+	Host getOrCreateHost(String name, CaseDbTransaction transaction) throws TskCoreException  {
+		
+		// must have a name
+		if (Strings.isNullOrEmpty(name) ) {
+			throw new IllegalArgumentException("Host name is required.");
+		}
+
+		CaseDbConnection connection = transaction.getConnection();
+
+		// First search for host by name
+		Optional<Host> host = getHost(name, connection);
+		if (host.isPresent()) {
+			return host.get();
+		}
+
+		// could'nt find it, create a new host
+		return createHost(name, connection);
+	}
+	
+	/**
+	 * Get host with given name. 
+	 * 
+	 * @param name Host name to look for. 
+	 * @param connection Database connection to use.
+	 * 
+	 * @return Optional with host.  Optional.empty if no matching host is found.
+	 * 
+	 * @throws TskCoreException 
+	 */
+	Optional<Host> getHost(String name, CaseDbConnection connection ) throws TskCoreException {
+		
+		String queryString = "SELECT * FROM tsk_hosts"
+							+ " WHERE LOWER(name) = LOWER('" + name + "')";
+
+		try (Statement s = connection.createStatement();
+				ResultSet rs = connection.executeQuery(s, queryString)) {
+
+			if (!rs.next()) {
+				return Optional.empty();	// no match found
+			} else {
+				return Optional.of(new Host(rs.getLong("id"), rs.getString("name")));
```

Comment:
I think we need to pass the status column value to the Host constructor as well.